aiidalab_qe/steps.py

```
@@ -856,6 +877,23 @@ def update_builder(buildy, resources, npools):
             electronic_type=ElectronicType(parameters["electronic_type"]),
         )
 
+        # Updating tot_charge of the builder
+        builder.relax.base.pw.parameters["SYSTEM"]["tot_charge"] = parameters[
+            "tot_charge"
+        ]
+        builder.bands.bands.pw.parameters["SYSTEM"]["tot_charge"] = parameters[
+            "tot_charge"
+        ]
+        builder.bands.scf.pw.parameters["SYSTEM"]["tot_charge"] = parameters[
+            "tot_charge"
+        ]
+        builder.pdos.scf.pw.parameters["SYSTEM"]["tot_charge"] = parameters[
+            "tot_charge"
+        ]
+        builder.pdos.nscf.pw.parameters["SYSTEM"]["tot_charge"] = parameters[
+            "tot_charge"
+        ]
```

I find this is error-prone. Can you try to using protocol override to override the base pw protocol and let it propagate to all other sub-workchain? Let me know if this is not clear and we can arrange a meeting to discuss.

---

> @AndresOrtegaGuerrero Thanks a lot for making the ball rolling. My comments are slightly general, I'll work in parallel to have a generic interface for similar parameter control widgets. What you can try in this PR in parallel is my above comment, moving this widget to "Advance settings" tab.

@unkcpz Thank you I will proceed then , One question , should I create then one specific class for this "Advanced Seetings" , since the idea will be to have Tot_charge, but I guess we want in this new tab to have other advanced settings like Periodicity (I submitted the PR in the aiida-quantumespresso) or Hubbard settings?

---

> 

@unkcpz  Maybe I would like if we can have a small talk regarding this one. I am not sure if this requires the modification of the workchain?

---

> One question , should I create then one specific class for this "Advanced Seetings" , since the idea will be to have Tot_charge, but I guess we want in this new tab to have other advanced settings like Periodicity.

Do you mean the class of widget for "Advance setting" specifically? I didn't look at the code, but maybe it is a good idea.